lib/modules.nix

```
@@ -1135,10 +1135,10 @@ rec {
         type = toType;
       });
       config = mkMerge [
-        {
-          warnings = optional (warn && fromOpt.isDefined)
+        (optionalAttrs warn {
+          warnings = optional fromOpt.isDefined
```

```suggestion
        (optionalAttrs (options?warnings) {
          warnings = optional (warn && fromOpt.isDefined)
```

This may be a little paranoid, but this way we don't make the top level value strict in `warn`.
The `options` attrset is computed before the `config`, so this is safe.

---

Done